Automattic/jetpack
# Jetpack: Add Jetpack Share Buttons block.

Fixes: https://github.com/Automattic/jetpack/issues/34050




#### Changes proposed in this Pull Request:

This Pull Request originated from this Hack Day project (p1HpG7-iVl-p2), which our team couldn't finish at that day specifically, but we combined a lot of knowledge into building a prototype, which could then be picked up in the future to build the actual solution. 

On March 2023 Hack Week, I picked up on this project to rise attention to it, since I wish for this feature myself. This is not yet finished, but it is on a much better stance and on a simple implementation path. It still relies on sharedaddy on the backend.

#### Other information:


#### Jetpack product discussion


p1HpG7-iVl-p2
pebfNN-6v-p2

#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Run the [extensions build ](https://github.com/Automattic/jetpack/tree/trunk/projects/plugins/jetpack/extensions)
* Edit any post (new or existing)
* Add a block, should be called **Sharing Buttons**
* Clicking the buttons should select services, save the post after selecting a few
* Preview/Visit the post to see the sharing buttons rendered where you added the block.

![image](https://user-images.githubusercontent.com/5550190/224868609-ab07a702-58cf-4a90-884a-fc1b38fd4907.png)

![image](https://user-images.githubusercontent.com/5550190/224868682-5fd859d3-cdca-4767-8bde-ff309d0a272a.png)
